mantidproject/mantid
# Neatenup RemoveBins and add tests

1. Added a missing test for RangeUnits
1. Consolidated cross parameter validation
1. Fixed a broken check for when units are the same
1. Only require the ws to have a unit when RangeUnits is set
1. Removed some units from the RangeUnits list as they made no sense

*No changes to how the guts of how removebins works!*

**To test:**

1. Test the validation of RemoveBins
 1. Xmax < Xmin
 1. WorkspaceIndex out of range
 1. Removals using RangeBins
 1. Removals using a workspace without units (use CreateWorkspace) with RangeUnits set to AsInput
 1. RangeUnits no longer contains the following entries ("Empty", "Label", "Time", "Degrees")

Re #18963.

### RELEASE NOTES

*Does not need to be in the release notes.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
